packages/core/src/components/graphs/area.ts

```diff
@@ -101,11 +101,15 @@ export class Area extends Component {
 				})}`
 			)
 			.node();
+		const sparklineColorObject = Tools.getProperty(
+			this.model.getOptions(),
+			"color",
+			"scale");
 		const colorValue = strokePathElement
 			? getComputedStyle(strokePathElement, null).getPropertyValue(
 					"stroke"
-			  )
-			: null;
+			)
+			: isGradientAllowed ? sparklineColorObject[Object.keys(sparklineColorObject)[0]] : null;
```

Comment:
very confusing code here. Are we able to clean this up? Also, shouldn't `isGradientAllowed` be check before checking the stroke?

Reply:
`isGradientAllowed` is now being checked before checking stroke.